src/Web/Controllers/AccountController.cs

```diff
@@ -243,11 +239,20 @@ private async Task<IActionResult> CallCustomAssessmentApi(CustomAssessmentViewMo
                 .Replace("@deviceFingerprintingId", model.DeviceFingerPrinting.SessionId)
                 .Replace("@deviceIpAddress", _contextAccessor.HttpContext.Connection.RemoteIpAddress.MapToIPv4().ToString());
             var assessment = new CustomAssessment { ApiName = model.ApiName, Payload = payload };
+
             var useV2 = model.Version.Equals(EndpointVersion.V2);
-            object response = useV2 ?
-                await _fraudProtectionService.PostAssessment(assessment, correlationId, envId) :
-                await _fraudProtectionService.PostCustomAssessment(assessment, correlationId, envId);
-            var fraudProtectionIO = new FraudProtectionIOModel(correlationId, payload, response, "Custom Assessment", true);
+            FraudProtectionIOModel fraudProtectionIO;
+            if (useV2)
```

Comment:
had to expand the tertiary to if/else as C# couldn't tell the generic type is similar enough to not matter

Reply:
Can  you please eloborate - why was this change needed?

Reply:
Yep @irinatarn ... Otherwise C# complains it can't find a compatible type for these two calls:

![image](https://github.com/microsoft/Dynamics-365-Fraud-Protection-Samples/assets/165767/c1841c1f-0927-4bc1-8074-dcf073e4ba54)

It's right.  It's just that it doesn't know it won't matter... so I've appeased it by breaking out the tertiary into if/else blocks.